### PR TITLE
Update envoyproxy/go-control-plane dependency

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

Update go-control-plane dependency

### CODE REVIEW

**Comment by howardjohn** [comment]
probably need https://github.com/envoyproxy/envoy/pull/22804

**Comment by jacob-delgado** [comment]
That PR just got merged and the go-control-plane repo was just updated.

**Review by howardjohn — APPROVED** [review]
thanks!

File: pilot/pkg/networking/core/v1alpha3/route/route_test.go [hunk 1]
```diff
@@ -951,7 +951,7 @@ func TestBuildHTTPRoutes(t *testing.T) {
 		g.Expect(ok).NotTo(gomega.BeFalse())
 		g.Expect(redirectAction.Redirect.ResponseCode).To(gomega.Equal(envoyroute.RedirectAction_PERMANENT_REDIRECT))
 		g.Expect(len(routes[0].ResponseHeadersToAdd)).To(gomega.Equal(1))
-		g.Expect(routes[0].ResponseHeadersToAdd[0].Append.Value).To(gomega.BeFalse())
```
Comment (jacob-delgado) [inline pilot/pkg/networking/core/v1alpha3/route/route_test.go:954]: The linter was failing due to using a deprecated api

**Pull request merged**